lib/rmt/mirror.rb

```diff
@@ -21,18 +22,30 @@ def mirror
   protected
 
   def mirror_metadata
-    temp_dir = Dir.mktmpdir
+    begin
+      local_repo_dir = File.join(@mirroring_base_dir, @local_path)
+      FileUtils.mkpath(local_repo_dir) unless Dir.exist?(local_repo_dir)
+    rescue StandardError => e
+      raise RMT::Mirror::Exception.new("Can not create a local repository directory: #{e}")
+    end
+
+    begin
+      temp_dir = Dir.mktmpdir
+    rescue StandardError => e
+      raise RMT::Mirror::Exception.new("Can not create a temporary directory: #{e}")
+    end
 
     @downloader = RMT::Downloader.new(
-      @repository_url,
-      temp_dir.to_s,
-      @logger
+      repository_url: @repository_url,
+      local_path: temp_dir.to_s,
+      logger: @logger
     )
 
     begin
       @downloader.download('repodata/repomd.xml.key')
       @downloader.download('repodata/repomd.xml.asc')
-    rescue RMT::Downloader::Exception # rubocop:disable Lint/HandleExceptions
+    rescue RMT::Downloader::Exception
+      @logger.info('Repository metadata signatures are missing')
```

Comment:
Those files are repository metadata signatures, and unsigned repositories exist. And these signatures aren't referenced in the XML.